File: 1-Part/3-chapter/practice/concat.js
function concat(array) {
    if (array.length === 0) {
        return ''
    } else if (array.length === 1) {
        return array[0]
    } else {
        let head = array[0]
        let tail = array.slice(1)
        return head + concat(tail)
    }
}

console.log(concat(['Hello', 'world']))

File: 1-Part/3-chapter/practice/concat.py
def concat(array):
    # Базовый случай: если массив пуст, вернуть пустую строку
    if len(array) == 0:
        return ''
    elif len(array) == 1:
        return array[0]  # Вернуть единственный элемент
    else:
        head = array[0]
        tail = array[1:]
        return head + concat(tail)


print(concat(['hello', 'world']))
